feat: confirmation before any action.

hemant-kr-meena opened the pull request:
## ⚡ PowerGrid - Pull Request

Welcome and thank you for your interest in contributing to our project!. You must use this template to submit a Pull Request or it will not be accepted.

--- 
#### Motivation

- [ ] Bug fix
- [ ] Enhancement
- [x] New feature
- [ ] Breaking change

#### Description

**Feature:** 
This PR adds a confirmation popup before executing any action via action buttons.
**Purpose:** 
To prevent unintended actions and improve user experience by adding an extra layer of confirmation.


As we know we use macros for additional functionality to add id and trigger dispatch events. In the same way, we can add wire:confirm attribute to buttons.

[Read more Livewire wire:confirm](https://livewire.laravel.com/docs/wire-confirm)

#### Example:
- Implemented simple confirmation using the following syntax:
```php
Button::add('delete')->id()->confirm("Are you sure?")
```
- Implemented confirmation prompts using the following syntax:
```php
Button::add('delete')->id()->confirmPrompt("Are you sure you want to delete ?", "Confirm")
```
#### Result:

<img width="361" alt="confirm" src="https://github.com/Power-Components/livewire-powergrid/assets/38046347/065fe796-9a41-4634-97e1-0fa6ec8501b7">

<img width="412" alt="confrim-prompt" src="https://github.com/Power-Components/livewire-powergrid/assets/38046347/c30e33cf-d650-407e-a583-e969a9cb2f9e">


#### Documentation

 This PR requires [Documentation](https://github.com/Power-Components/powergrid-doc) update?

- [x] Yes
- [ ] No
- [x] I have already submitted a Documentation pull request.

hemant-kr-meena committed:
adding English translations for default message

hemant-kr-meena committed:
writing tests for confirm and confirmPrompt method

hemant-kr-meena committed:
adding macros functions and methods for Button class

hemant-kr-meena committed:
fix: test ConfirmPromptTest.php

hemant-kr-meena committed:
style: adding line break in both macro functions

luanfreitasdev commented on resources/lang/en/datatable.php:
```diff
@@ -64,4 +64,12 @@
     'multi_sort' => [
         'message' => 'Multiple sort is active',
     ],
+    'buttons-macros' => [
```
Thanks for the PR! You can add these lines to other languages as well?

hemant-kr-meena replied:
Task has been completed [62f03df](https://github.com/Power-Components/livewire-powergrid/pull/1599/commits/62f03dfcc90742b1c514e9abb3851f445d1bd693)

luanfreitasdev commented on src/Components/Actions/Macros.php:
```diff
@@ -194,5 +194,27 @@ public static function boot(): void
+            $this->dynamicProperties['confirmPrompt'] = [
+                'component' => 'button',
+                'attribute' => 'wire:confirm.prompt',
+                'value'     => "$message |$confirm_value",
```
Use camelCase

hemant-kr-meena replied:
camelCase for $**confirm_value** ?

so did I also need to change the translations placeholder for `:confirm_value` to use **camelCase** `:confirmValue`.

luanfreitasdev reviewed (changes requested):
Can you add a doc about this? in https://livewire-powergrid.com/table-features/button-class.html#button-methods

hemant-kr-meena commented:
Thank you for your feedback. I will make the requested changes and submit a docs pull request to the docs repo within a few days.

hemant-kr-meena committed:
use camel case for $confirm_value to $confirmValue

hemant-kr-meena committed:
adding  missing multi_sort translations

hemant-kr-meena committed:
adding english lang messages in all language

hemant-kr-meena committed:
wip

we change lang translation separator(-) to underscore(-) to make it similar to all other translations.

luanfreitasdev commented:
Thank you for this PR!

hemant-kr-meena commented:
Hi @luanfreitasdev, I marked all requested changes as resolved. But in PR it still shows that one change is requested. Why is this showing this?
 
 Did I need to make any other changes regarding this PR?

luanfreitasdev committed:
pint
